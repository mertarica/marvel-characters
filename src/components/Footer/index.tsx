import React from "react";
import Container from "react-bootstrap/Container";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";

const Footer = () => {
  const currentYear = new Date().getFullYear();
  return (
    <footer className="c-footer" data-testid="footer">
      <Container fluid>
        <Row>
          <Col xs={12}>
            <span className="c-footer__brand">Â© {currentYear}</span>
          </Col>
        </Row>
      </Container>
    </footer>
  );
};

export default Footer;
