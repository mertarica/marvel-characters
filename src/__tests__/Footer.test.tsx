import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Footer from "../components/Footer";

describe("<Footer />", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders without crash", () => {
    render(<Footer />);
    expect(screen.getByTestId("footer")).toBeInTheDocument();
  });

  it("renders correct year", () => {
    render(<Footer />);
    expect(screen.getByTestId("footer")).toHaveTextContent(
      String(new Date().getFullYear())
    );
  });
});
