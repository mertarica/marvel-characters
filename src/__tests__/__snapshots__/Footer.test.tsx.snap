// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders correctly 1`] = `
<footer
  className="c-footer"
  data-testid="footer"
>
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <span
          className="c-footer__brand"
        >
          Â© 
          2022
        </span>
      </div>
    </div>
  </div>
</footer>
`;
